dart/lib/src/protocol/message.dart

```diff
@@ -11,14 +11,14 @@ import 'package:meta/meta.dart';
 @immutable
 class Message {
   /// The fully formatted message. If missing, Sentry will try to interpolate the message.
-  final String formatted;
+  final String? formatted;
```

Comment:
this is required https://develop.sentry.dev/sdk/event-payloads/message/

Reply:
I would leave it as optional because it gets filled from nullable fields, such as `SentryClient.captureMessage(...)` and the message could be null.

Reply:
SentryClient.captureMessage(null) should not raise an event at all, it's not going to be useful.
if we create a Message without a `formatted` field, it'll cause a backend error during ingestion.

Reply:
just reforcing this, captureMessage(String?) is fine, but we'll bail out before creating a Message, so `formatted` should be non nulable to not generate a backend error

Reply:
I think captureMessage sholdn't take null

Reply:
this would give a lot of boilerplate or force an error on the users code, think about?

final str = getthisfromApi();
// str is nullable
Sentry.captureMessage(str!); // could cause an error on app's code.

// doing this would be annoying
if (str != null) {
  Sentry.captureMessage(str!);
}

Reply:
as a `Message` is created only after capturing, hence the field `formatted` should not be nullable